### DISCUSSION TITLE
fixed point delta time ?

### DISCUSSION THREAD

**Discussion by SeleDreams** [SanderMertens/flecs#1377] [Q&A]
I am working on a game engine for very low end devices like the DS. I wanted to use flecs, but I noticed that if i redefine ecs_ftime_t as an int, the engine will freeze on a call to ecs_progress.
floating point operations are a no-go for this kind of platform.
Is there a way to completely skip the deltatime aspect of flecs ? I just want to use it for ecs. I already manage myself the delta aspect of my engine. I tried passing manual values such as 1 but it doesn't seem to do much.

**Comment by copygirl** [comment]
Relevant issue: #1166

The issue you described should've been fixed already, but perhaps there was a regression? Worth making sure you are on a recent flecs version. You also do not have to use `delta_time` yourself, just call `ecs_progress` at determined intervals and treat each step as progressing by one frame. Or use a component / static value to keep track of delta time and time since world start or similar. If a workaround works for you.

**Reply by SeleDreams** [reply]
I have my own deltatime used by my engine, that's how it calls ecs_progress every frame at 60fps. 
i've set ecs_ftime_t as int and I called ecs_progress as
`ecs_progress(p_scene->world, 1);`
however ecs_progress after a single frame never returns.

I noticed that if i set a breakpoint and continue through each frame, it will actually work, but if i just play it normally at 60 fps it will freeze after 1 or 2 frames.
there must be some timing management code going on, but i don't want it

**Reply by SeleDreams** [reply]
I found a fix.
After the initialization of the world, I had to add the line `ecs_measure_frame_time(p_scene->world,false);` to disable the time management of flecs. after that it didn't freeze anymore

**Reply by copygirl** [reply]
That might've been enabled by the REST addon? I wonder if it's necessary for statistics to work.

Worth creating an issue for. This code might be the culprit?
https://github.com/SanderMertens/flecs/blob/dd39321f415747cde7e5a73d8a4d0d7c7bdf90f6/src/world.c#L1843-L1866

**Reply by SeleDreams** [reply]
that's at least around here that it stayed stuck before
